configuration/config2/XSDValueSetParser.hpp

```diff
@@ -0,0 +1,56 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#ifndef _CONFIG2_XSDVALUESETPARSER_HPP_
+#define _CONFIG2_XSDVALUESETPARSER_HPP_
+
+#include <string>
+#include <memory>
+#include <map>
+#include <boost/property_tree/ptree.hpp>
+
+#include "XSDConfigParser.hpp"
+#include "ConfigItemValueSet.hpp"
+
+namespace pt = boost::property_tree;
+
+class XSDValueSetParser : public XSDConfigParser
+{
+    public:
+
+        XSDValueSetParser(const std::string &basePath, std::shared_ptr<ConfigItem> pValueSet) : 
+            XSDConfigParser(basePath, pValueSet) { m_pValueSet = std::dynamic_pointer_cast<ConfigItemValueSet>(pValueSet);  }
+        virtual ~XSDValueSetParser() { }
+        virtual void parseXSD(const pt::ptree &valueSetTree);
+        void parseAttributeGroup(const pt::ptree &attributeTree);
+        void parseAttribute(const pt::ptree &attr);
+    
+
+    protected:
+
+        XSDValueSetParser() { }
+
+
+    protected:
+
+        std::shared_ptr<ConfigItemValueSet> m_pValueSet;
+
+};
+
+
```

Comment:
you'll want to remove all superfluous new lines

Reply:
OK. I figured there would be a massive cleanup before the final pull request anyway.